electrum/gui/qt/paytoedit.py

```diff
@@ -78,336 +67,169 @@ def __init__(self):
         self.verticalMargins += documentMargin * 2
         self.heightMin = self.fontSpacing + self.verticalMargins
         self.heightMax = (self.fontSpacing * 10) + self.verticalMargins
+        self.single_line = True
         self.update_size()
 
     def update_size(self):
         docLineCount = self.document().lineCount()
-        docHeight = max(3, docLineCount) * self.fontSpacing
+        docHeight = max(1 if self.single_line else 3, docLineCount) * self.fontSpacing
         h = docHeight + self.verticalMargins
         h = min(max(h, self.heightMin), self.heightMax)
         self.setMinimumHeight(int(h))
         self.setMaximumHeight(int(h))
-        self.verticalScrollBar().setHidden(docHeight + self.verticalMargins < self.heightMax)
+        if self.single_line:
+            self.setHorizontalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
+            self.setLineWrapMode(QTextEdit.LineWrapMode.NoWrap)
+        else:
+            self.setHorizontalScrollBarPolicy(Qt.ScrollBarAsNeeded)
+            self.verticalScrollBar().setHidden(docHeight + self.verticalMargins < self.heightMax)
+            self.setLineWrapMode(QTextEdit.LineWrapMode.WidgetWidth)
 
+class PayToEdit(QObject, Logger, GenericInputHandler):
 
-class PayToEdit(Logger, GenericInputHandler):
+    paymentIdentifierChanged = pyqtSignal()
 
     def __init__(self, send_tab: 'SendTab'):
+        QObject.__init__(self, parent=send_tab)
         Logger.__init__(self)
         GenericInputHandler.__init__(self)
-        self.line_edit = QLineEdit()
+
         self.text_edit = ResizingTextEdit()
-        self.text_edit.hide()
+        self.text_edit.textChanged.connect(self._handle_text_change)
```

Comment:
The split might have been intentional, but I didn't see any functional reason why. Technically it might have had a reason, but after refactoring this approach didn't fit well and it lead to complications keeping these in sync (in combination with textChanged events).

Reply:
> Note that there is a toolbar menu in the Send tab (near top right corner) that includes a pay-to-many checkbox (`send_tab.paytomany_menu`). Now with these changes, the checkbox state can easily become out-of-sync.

Not sure what the idea of the menu toggle is. Should this override the size regardless of the contents (i.e. follow content, until user overrides using the toggle?)

Reply:
I think one of the goals in https://github.com/spesmilo/electrum/commit/1f4cedf56a342ed4b1e1e5dea69e037dd1f8083f was that by default the input field is only a single line, pressing enter is suppressed, and it is using the menu toggle (or explicit pasting (ctrl+v or using button)) that you can change to multiline. Though I think this change was subjective.
I think the main reason @ecdsa split the textedit into lineedit+textedit was the "visual appearance" remark in the commit message: on certain linux styles, one-line QTextEdits look significantly different from QLineEdits, and @ecdsa wanted all the input fields on the Send tab to have a homogene look.

Reply:
> on certain linux styles, one-line QTextEdits look significantly different from QLineEdits, and @ecdsa wanted all the input fields on the Send tab to have a homogeneous look.

I see.. I've checked most of the styles that come pre-installed with debian kde, and looked at a windows portable build on windows 11. There is indeed a slight difference on linux with some styles, but it's rather minor IMHO. I'd rather not refactor back to the two alternating edit controls, as it's a maintenance hazard..

I can try fiddling with the styling of the QTextEdit, or I can define a custom edit control deriving from QTextEdit and use that also for single line edit controls, so we're sure to have the same styling everywhere. In any case, I don't want to spend too much time on it :)